platforms/software/build-init/src/main/java/org/gradle/buildinit/plugins/internal/ProjectLayoutSetupRegistry.java

```diff
@@ -54,16 +54,43 @@ public TemplateOperationFactory getTemplateOperationFactory() {
         return templateOperationFactory;
     }
 
+    // Currently used by `build-logic/build-init-samples/src/main/kotlin/gradlebuild/samples/SamplesGenerator.kt`
+    @SuppressWarnings("unused")
+    public List<Language> getLanguagesFor(ComponentType componentType) {
+        List<BuildGenerator> generators = getGeneratorsFor(componentType);
+
+        List<Language> result = new ArrayList<>();
+        for (Language language : Language.values()) {
+            for (BuildGenerator generator : generators) {
+                if (generator.productionCodeUses(language)) {
+                    result.add(language);
+                    break;
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Returns the component types, in display order.
+     */
     public List<ComponentType> getComponentTypes() {
-        return ImmutableList.copyOf(ComponentType.values());
+        return Arrays.asList(ComponentType.values());
+    }
+
+    /**
+     * Returns the default component type to use for interactive initialization.
+     */
+    public ComponentType getDefaultComponentType() {
+        return getComponentTypes().get(0);
```

Comment:
❔ Would it better to be explicit here with `ComponentType.APPLICATION`, since it's a enum?

Reply:
I've changed this